go/src/koding/kites/config/cache.go

```diff
@@ -34,19 +35,39 @@ type Cache struct {
 // If it was not possible to create or open BoltDB database,
 // an in-memory cache is created.
 func NewCache(options *CacheOptions) *Cache {
+	db, err := newBoltDB(options)
+	if err != nil {
+		log.Println(err)
+	}
+
 	return &Cache{
-		EncodingStorage: storage.NewEncodingStorage(newBoltDB(options), options.Bucket),
+		EncodingStorage: storage.NewEncodingStorage(db, options.Bucket),
 	}
 }
 
-func newBoltDB(o *CacheOptions) *bolt.DB {
-	os.MkdirAll(filepath.Dir(o.File), 0755)
+// NewCache returns new cache value backed by BoltDB.
+func NewBoltCache(options *CacheOptions) (*Cache, error) {
+	db, err := newBoltDB(options)
+	if err != nil {
+		return nil, err
+	}
 
-	if db, err := bolt.Open(o.File, 0644, o.BoltDB); err == nil {
-		return db
+	bolt, err := storage.NewBoltStorageBucket(db, options.Bucket)
+	if err != nil {
+		return nil, err
 	}
 
-	return nil
+	return &Cache{
+		EncodingStorage: &storage.EncodingStorage{
+			Interface: bolt,
+		},
+	}, nil
+}
+
+func newBoltDB(o *CacheOptions) (*bolt.DB, error) {
+	os.MkdirAll(filepath.Dir(o.File), 0755)
```

Comment:
I think there's no need to duplicate `os` package file handling. `os.MkdirAll` is a best-effort attempt to create base directory, if it's missing. We ignore the error, since `bolt.Open` is going to report the very same error that os.MkdirAll would:

- if there are no permissions to create dir with `os.MkdirAll`, `bolt.Open` will fail with `2016/10/20 15:33:01 open abc/abc.bolt: permission denied`
- if os.MkdirAll failed for whatever other reason, `bolt.Open` will just fail with `2016/10/20 15:32:38 open abc/abc/abc.bolt: no such file or directory`, which is pretty self-describing upon inspection.

If that's not enough, how would you handle it differently?